docs/data/date-pickers/adapters-locale/CompactFieldFormat.js

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import dayjs from 'dayjs';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DateField } from '@mui/x-date-pickers/DateField';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+
+export default function CompactFieldFormat() {
+  return (
+    <LocalizationProvider dateAdapter={AdapterDayjs}>
+      <DemoContainer components={['DateField', 'DatePicker']}>
+        <DateField
+          label="Date Field"
+          defaultValue={dayjs('2022-04-17')}
+          formatDensity="dense"
```

Comment:
What do you think about using a bit different names to better align with what the user would expect and potentially easily change the default to the `regular` or `normal` one in v7?
The suggested density formats are aligned with how our field components are currently built, but the reality is that in the real world, it causes more issues than it solves and users might expect that `regular`/`normal` density would be the usual date format without any spacing, whereas we are inverting this logic the other way around.

My suggestion would be to go with `regular` or `normal` and `sparse` or anything better you can come up with as the current default.

My idea in other words - look at the formats from the perspective of a user used to date formats having no spacing and not from the perspective of our case, where we added spacing where usually there was none.

Reply:
Alternatively we can go so `dense` and `sparse`.
That way we don't have a `regular` / `normal` that is very subjective depending on what people find "normal".

I would not rely to much on the fact that "regular = we do not touch the format".
Because in v7, if we go for the Spectrum approach with several HTML inputs.
Then this prop will be used to decide the spacing between those inputs and the separators, so the notion of "we do not touch the format" won't make sense.